niworkflows/viz/plots.py

```diff
@@ -244,28 +257,57 @@ def plot_carpet(
         legend = False
 
     else:  # Volumetric NIfTI
-        img_nii = check_niimg_4d(img, dtype="auto",)
+        # Load data
+        img_nii = check_niimg_4d(
+            img,
+            dtype="auto",
+        )
         func_data = _safe_get_data(img_nii, ensure_finite=True)
         ntsteps = func_data.shape[-1]
-        data = func_data[atlaslabels > 0].reshape(-1, ntsteps)
-        oseg = atlaslabels[atlaslabels > 0].reshape(-1)
 
-        # Map segmentation
+        # Dilate brain mask
+        crown_mask, func_seg_mask = get_dilated_brainmask(atlaslabels, brainmask)
```

Comment:
Here you could do:

```suggestion
        crown_mask, func_seg_mask = get_dilated_brainmask(np.bool_(atlaslabels), brainmask)
```

Or

```suggestion
        crown_mask, func_seg_mask = get_dilated_brainmask(atlaslabels != 0, brainmask)
```